web/src/features/map/Map.tsx

```diff
@@ -153,14 +156,54 @@ export default function MapPage({ onMapLoad }: MapPageProps): ReactElement {
 
   useEffect(() => {
     // Run on first load to center the map on the user's location
-    if (!map || isError || !isFirstLoad) {
+    if (!map || isError || !isFirstLoad || !isSourceLoaded || !data) {
       return;
     }
     if (data?.callerLocation && !selectedZoneId) {
       map.flyTo({ center: [data.callerLocation[0], data.callerLocation[1]] });
+
+      const handleIdle = () => {
+        if (
+          map.isSourceLoaded(ZONE_SOURCE) &&
+          map.areTilesLoaded() &&
+          data?.callerLocation
+        ) {
+          const source = map.getSource(ZONE_SOURCE);
+          const layer = map.getLayer('zones-clickable-layer');
+          if (!source) {
+            console.error(`Source "${ZONE_SOURCE}" not found`);
+            return;
+          }
+          if (!layer) {
+            console.error('Layer "zones-clickable-layer" not found or not rendered');
+            return;
+          }
+          const zoneFeature = getZoneIdFromLocation(
+            map,
+            data?.callerLocation,
```

Comment:
We should ensure callerLocation actually exist and has valid coordinates before passing it along here. There have been Sentry issues where the things have failed because the coords where non valid. (I don't know exactly how it happened yet though)